syntax = "proto3";

package student;

service StudentService {
  rpc CreateStudent(CreateStudentRequest) returns (StudentResponse);
  rpc GetStudent(GetStudentRequest) returns (StudentResponse);
  rpc ListStudents(ListStudentsRequest) returns (ListStudentsResponse);
  rpc UpdateStudent(UpdateStudentRequest) returns (StudentResponse);
  rpc DeleteStudent(DeleteStudentRequest) returns (DeleteStudentResponse);
}

message Student {
  int32 id = 1;
  string nome = 2;
  string email = 3;
  string matricula = 4;
}

message CreateStudentRequest {
  string nome = 1;
  string email = 2;
  string matricula = 3;
}

message GetStudentRequest {
  int32 id = 1;
}

message ListStudentsRequest {}

message ListStudentsResponse {
  repeated Student students = 1;
}

message UpdateStudentRequest {
  int32 id = 1;
  string nome = 2;
  string email = 3;
  string matricula = 4;
}

message DeleteStudentRequest {
  int32 id = 1;
}

message DeleteStudentResponse {
  bool success = 1;
}

message StudentResponse {
  Student student = 1;
}
